io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java

```diff
@@ -0,0 +1,316 @@
+)
+@EventTopics({ //
+		EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE, //
+})
+
+public class EvcsWebastoNextImpl extends AbstractOpenemsModbusComponent
+		implements EvcsWebastoNext, Evcs, ManagedEvcs, ModbusComponent, OpenemsComponent, EventHandler {
+
+	private final Logger log = LoggerFactory.getLogger(EvcsWebastoNext.class);
+
+	private Config config = null;
+
+	private static final int DEFAULT_LIFE_BIT = 1;
+
+	/**
+	 * Handles charge states.
+	 */
+	private final ChargeStateHandler chargeStateHandler = new ChargeStateHandler(this);
+
+	/**
+	 * Processes the controller's writes to this evcs component.
+	 */
+	private final WriteHandler writeHandler = new WriteHandler(this);
+
+	public EvcsWebastoNextImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				ModbusComponent.ChannelId.values(), //
+				Evcs.ChannelId.values(), //
+				ManagedEvcs.ChannelId.values(), //
+				EvcsWebastoNext.ChannelId.values());
+	}
+
+	@Reference
+	private EvcsPower evcsPower;
+
+	@Reference
+	protected ConfigurationAdmin cm;
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	protected void setModbus(BridgeModbus modbus) {
+		super.setModbus(modbus);
+	}
+
+	@Activate
+	private void activate(ComponentContext context, Config config) throws OpenemsNamedException {
+		this.config = config;
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+
+		/*
+		 * Calculates the maximum and minimum hardware power dynamically by listening on
+		 * the fixed hardware limit and the phases used for charging
+		 */
+		Evcs.addCalculatePowerLimitListeners(this);
+
+		this.applyConfig(context, config);
+	}
+
+	@Modified
+	private void modified(ComponentContext context, Config config) throws OpenemsNamedException {
+		if (super.modified(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		this.applyConfig(context, config);
+	}
+
+	private void applyConfig(ComponentContext context, Config config) {
+		this.config = config;
+		this._setChargingType(ChargingType.AC);
+		this._setFixedMinimumHardwarePower(config.minHwCurrent() / 1000 * 3 * 230);
+		this._setFixedMaximumHardwarePower(config.maxHwCurrent() / 1000 * 3 * 230);
+		this._setPowerPrecision(230);
+
+	}
+
+	@Override
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		// Cannot read the gaps, therefore there are so many tasks
+		var modbusProtocol = new ModbusProtocol(this, //
+				new FC3ReadRegistersTask(1000, Priority.HIGH,
+						m(EvcsWebastoNext.ChannelId.CHARGE_POINT_STATE, new UnsignedWordElement(1000)), //
+						m(ManagedEvcs.ChannelId.CHARGE_STATE, new UnsignedWordElement(1001)), //
```

Comment:
Remove. Channel set automatically by the WriteHandler
Noted in the Channel description